src/modules/sidebar/mixins/shareAPI.js

```diff
@@ -1,82 +1,56 @@
 import axios from '@nextcloud/axios'
 import { generateUrl } from '@nextcloud/router'
-import { showError, showSuccess, showWarning } from '@nextcloud/dialogs'
+import { showSuccess } from '@nextcloud/dialogs'
 import '@nextcloud/dialogs/dist/index.css'
+import displayError from '../../../shared/utils/displayError.js'
 
 export default {
 	methods: {
 		async getSharedWithFromBE() {
 			try {
 				const res = await axios.get(generateUrl('/apps/tables/share/table/' + this.activeTable.id))
-				if (res.status !== 200) {
-					showWarning(t('tables', 'Sorry, something went wrong.'))
-					console.debug('axios error', res)
-				}
-				// this.sharedWith = res.data.sort((a, b) => { return a.userReceiver.localeCompare(b.userReceiver) })
 				return res.data
 			} catch (e) {
-				console.error(e)
-				showError(t('tables', 'Could not load shares from back end'))
+				displayError(e, t('tables', 'Could not fetch shares.'))
 			}
 		},
 
 		async sendNewShareToBE(share) {
+			const data = {
+				nodeType: 'table',
+				nodeId: this.activeTable.id,
+				receiver: share.user,
+				receiverType: (share.isNoUser) ? 'group' : 'user',
+				permissionRead: true,
+				permissionCreate: true,
+				permissionUpdate: true,
+				permissionDelete: false,
+				permissionManage: false,
+			}
+			let tableId = null
 			try {
-				const data = {
-					nodeType: 'table',
-					nodeId: this.activeTable.id,
-					receiver: share.user,
-					receiverType: (share.isNoUser) ? 'group' : 'user',
-					permissionRead: true,
-					permissionCreate: true,
-					permissionUpdate: true,
-					permissionDelete: false,
-					permissionManage: false,
-				}
-				// console.debug('data array', data)
 				const res = await axios.post(generateUrl('/apps/tables/share'), data)
-				if (res.status === 200) {
-					await this.$store.dispatch('setTableHasShares', { tableId: res.data.nodeId, hasSHares: true })
-				} else {
-					showWarning(t('tables', 'Sorry, something went wrong.'))
-					console.debug('axios error', res)
-					return false
-				}
-				// console.debug('new share was saved', res)
+				tableId = res.data.nodeId
 				showSuccess(t('tables', 'Saved new share with "{userName}".', { userName: res.data.receiverDisplayName }))
 			} catch (e) {
-				console.error(e)
-				showError(t('tables', 'Could not create new share'))
+				displayError(e, t('tables', 'Could not create share.'))
 			}
+			await this.$store.dispatch('setTableHasShares', { tableId, hasShares: true })
```

Comment:
Cool... exceptions from the dispatch will not be caught anymore.

But the dispatch will happen even when network exceptions are caught. It will not work because tableId will be null - but I think it would be better to return from the catch block so the dispatch does not cause another error.

Reply:
Hehe... I'm just crazy lucky. :wink: 
This is the only case i looked at. :dart: